tests/data/parsers/test_parser_classification_rule.py

```diff
@@ -220,19 +224,51 @@ def test_feedback_for_criteria_with_gaps_and_overlap(
                 [2, "<0", "<0", ">10"],
                 [3, "-", "-", "-"],
             ],
-            """For conditions: (varA: <0, varB: <0). Gap for variable varC in range -inf:0.0\nFor conditions: (varA: <0). Gap for variable varB in range 0.0:inf\nOverlap for variable varA in range -inf:0.0""",
+            [
+                call(
+                    """For conditions: (varA: <0, varB: <0). Gap for variable varC in range -inf:0.0\nFor conditions: (varA: <0). Gap for variable varB in range 0.0:inf\nOverlap for variable varA in range -inf:0.0"""
+                )
+            ],
+        ),
+        (
+            [
+                ["output", "MIN_water_depth_mNAP", "MAX_flow_velocity", "MAX_chloride"],
+                [1, "<0.10", "-", ">300"],  # to dry
+                [2, "<0.10", "-", "< 400"],  # also to dry
+                [3, ">4.0", ">3.0", "-"],  # to deep
+                [4, ">4.0", "<2.0", "-"],  # also to deep
+                [5, "-", "-", ">400"],  # to salty
+                [6, "-", ">1.5", "-"],  # to fast flowing
+                [7, "-", ">1.5", ">300"],  # alos to fast flowing
+                [8, "0.20:4.0", "0.0:1.5", "0:400"],  # perfect for aquatic plants
+            ],
+            [
+                call(
+                    "For conditions: (MIN_water_depth_mNAP: -, MAX_flow_velocity: -). Gap for variable MAX_chloride in range -inf:400.0\nFor conditions: (MIN_water_depth_mNAP: -, MAX_flow_velocity: >1.5). Overlap for variable MAX_chloride in range 300.0:inf\nFor conditions: (MIN_water_depth_mNAP: 0.20:4.0, MAX_flow_velocity: 0.0:1.5). Gap for variable MAX_chloride in range -inf:0.0\nFor conditions: (MIN_water_depth_mNAP: 0.20:4.0, MAX_flow_velocity: 0.0:1.5). Gap for variable MAX_chloride in range 400.0:inf\nFor conditions: (MIN_water_depth_mNAP: <0.10, MAX_flow_velocity: -). Overlap for variable MAX_chloride in range 300.0:400.0\nFor conditions: (MIN_water_depth_mNAP: -). Overlap for variable MAX_flow_velocity in range 1.5:inf"
```

Comment:
same remark about missing periods

Reply:
I don't understand the ordering of the warnings. Not a real problem, but I do wonder where it comes from?

Reply:
I also find understanding the warning, and relating it to the given table, quite difficult. I expect followup tickets in the future from confused users.

Reply:
ordering of the warnings... 
I loop from right to left through the variables. Checking first the most right column. But filtered on unique values in the other columns (let's call these chunks). So eg three columns (varA, varB, varC) will give first a list with all warnins for chunks of varC: (For conditions: (VarA, varB). Gap for varC ... ). Then going to varB (For conditions: (varA..). Gap for VarB) and then to varA. 
Per chunk the values are sorted, and that gives the order of the warnings, because you loop over these values and checking them with the previosu values. *sigh

Reply:
and for understanding the warning. It might be better when the output is also included in the sentence. This came up when talking to Marc as well, but I parked it for now, because it was quite tricky to get the right output in there, and even more trickier to get all the output values involved in such conditions in there..
also I'm afraid that the warnings become even longer when adding that.
I think we should open indeed a new ticket to make this a  bit more clearer, but I prefer with input from users/Marc